src/pal/src/map/map.cpp

```diff
@@ -496,7 +537,22 @@ CorUnix::InternalCreateFileMapping(
         //
         
         /* Anonymous mapped files. */
-        if (strcpy_s(pImmutableData->szFileName, sizeof(pImmutableData->szFileName), "/dev/zero") != SAFECRT_SUCCESS)
+        if (pImmutableData->lpFileName != NULL)
```

Comment:
`_ASSERTE(pImmutableData->lpFileName == NULL)` without the error code path would be enough here, this condition would indicate a bug in the PAL and cannot be triggered by incorrect arguments passed to the PAL functions.